### PR TITLE
Add helper for standard voting input parameters, add Parameters class

### PR DESCRIPTION
**Pull request by mashton**
* Adds a helper for defining standard voting input parameters, as discussed with @shaunagm.
* Pass parameters to `process.start(params)` as a class object with dynamically defined attributes, instead of a dict. 
* I also consolidated most of the relevant imports for plugin into one file (`plugin_manager.py`) so that plugin authors don't need to import from multiple files.

we dont have test coverage for most, so will test manually..
- [x] test slack vote
- [x] test discourse vote
- [x] test loomio vote
- [ ] test github vote
- [ ] test oc vote

### CODE REVIEW

File: metagov/metagov/core/plugin_manager.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        if extra_properties:
+            for (prop, definition) in extra_properties.items():
+                properties[prop] = definition
+
+        schema = {"properties": properties}
+        if required:
+            schema["required"] = [prop for prop in required if prop in properties.keys()]
+
+        return schema
```
Comment (mashton) [inline metagov/metagov/core/plugin_manager.py:53]: new `VotingStandard` class

File: metagov/metagov/plugins/discourse/models.py [hunk 2]
```diff
@@ -265,20 +265,15 @@ def process_discourse_webhook(self, request):
...
-
-    input_schema = {
-        "type": "object",
-        "properties": {
-            "title": {"type": "string"},
-            "options": {"type": "array", "items": {"type": "string"}},
-            "details": {"type": "string"},
+    input_schema = VotingStandard.create_input_schema(
+        include=["title", "options", "details", "closing_at"],
+        extra_properties={
```
Comment (mashton) [inline metagov/metagov/plugins/discourse/models.py:270]: Discourse updated to use VotingStandard

**Comment by shaunagm** [comment]
This looks good to me but I haven't done a deep dive. There's a bunch of extra stuff in the "files changed" - should I just ignore that, and focus on the parts that you've pulled out here?

Also, if you like I can test by trying to incorporate this into the Github votes.

**Comment by mashton** [comment]
Yeah the other edits mostly pertain to this:
> I also consolidated most of the relevant imports for plugin into one file (plugin_manager.py) so that plugin authors don't need to import from multiple files.

You can feel free to ignore this pieces or comment on it, it's mostly unrelated

**Review by shaunagm — APPROVED** [review]

**Pull request merged**